### ISSUE TITLE
Large percentage of unsuccessful processing types

### ISSUE THREAD

**Issue by JayVaz18** [nanoporetech/megalodon#138]
Does this mean that the majority (almost all) of the reads weren't properly processed? 

```
[11:48:06] Unsuccessful processing types:
    99.3% (   3971 reads) : Guppy server timeout (see --guppy-timeout argument)                             
     0.4% (     17 reads) : No alignment      

```

The command I ran to get 5mC in all contexts is below:

```
megalodon /Users/jakevazquez18/Desktop/guppy/fast5/megalodon/fast5s/0/ --outputs basecalls mappings mod_mappings mods per_read_mods --reference /Users/jakevazquez18/Desktop/Genomes/Clemence/w303.fa --mod-motif m C 0 --processes 30 --guppy-server-path /Users/jakevazquez18/Desktop/ont-guppy-cpu/bin/guppy_basecall_server --output-directory /Users/jakevazquez18/Desktop/guppy/fast5/megalodon/meg_results2/

```

What I would like to do is generate a table that has the position of the target modified base, the probability it was methylated and the read_id. I see the output file has a .bed file but the last column contains all 0.0 except for a couple of rows that contain 100.0.

**Comment by JayVaz18** [comment]
If you can explain the output and what the .bed file columns as well as the Sql .db columns represent that would help a ton

**Comment by marcus1487** [comment]
The guppy server error indicates that the requested compute resources cannot keep up with the basecalling load. For running in CPU resources the guppy timeout argument should be increased, likely to 500 or more. 

The output formats are explained in the docs here: https://nanoporetech.github.io/megalodon/file_formats.html

Direct access to the sql tables is not recommended or supported. Beta support is available through the API from the mods module. For access to the raw per read data the text format would be the recommended output.

**Comment by JayVaz18** [comment]
So, if I add the --guppy-timeout argument with a value of 500 to the command I have put in my first comment above it should work fine?

**Comment by marcus1487** [comment]
I can’t say for certain what value the timeout parameter should be for your compute environment. It may need to be even higher than 500. Also be sure you are using the most recent version of megalodon. There are some recent improvements to the multiprocessing. The defaults are designed for GPU processing.

**Comment by JayVaz18** [comment]
Oh okay. I just tried it using --guppy-timeout argument 2000 because 500 showed the same error just at a little bit of a later time than before. Using 2000 collected more reads but it still stopped. I am going to try on GPU because I'm not sure which value to set it on for all the reads to be processed.

**Comment by JayVaz18** [comment]
I still received the guppy server error even when using GPU. I'm not sure what I'm doing wrong

**Comment by marcus1487** [comment]
I would recommend turning down your processes setting as well. Can you provide the megalodon version? Also providing the compute environment details would be helpful.

**Comment by JayVaz18** [comment]
I will try that. I am using 2.3.1 of Megalodon in a python==3.8 Conda envs on a 32 GB macOS Mojave Desktop. I also have access to a remote connection to a Ubuntu.

**Comment by JayVaz18** [comment]
After running over the weekend, this was the output. A couple things jump out to me such as the "No Alignment" and the fact that 65% of reads were shown as Unsuccessful processing types and there was also no processing for "input queue capacity extract_signal, output queue capacity Basecalls, output queue capacity mappings, and output queue capacity per_read_mods - does this mean these were not processed at all?

<img width="1126" alt="Screen Shot 2021-05-10 at 8 40 57 AM" src="https://user-images.githubusercontent.com/72083618/117660674-722b7600-b16b-11eb-91d1-04000db680a1.png">

**Comment by marcus1487** [comment]
The 65% of reads showing no alignment indicate that 65% of reads do not map to the provided reference. This is indicated as an unsuccessful processing type as no reference anchored modified base calls can be made without a reference mapping. There are a number of reasons for this including but not limited to sample contamination, very short reads, very high modified base content.

For the input/output queue progress bars. These indicate progress during a run. Once the run is complete, all of the processing queues have been exhausted and are correctly presented as empty. These values are displayed in order to identify potential IO issues which could slow the processing of a run).

**Comment by JayVaz18** [comment]
Okay thanks. Last thing, do you know why I would be getting negative mod_prob values? I tried running Megalodon on a directory of lots of .fast5 files that have methylation and got results with negative mod_prob values.

**Comment by marcus1487** [comment]
In what output file are these negative mod_probs values found? How negative are these values? Could it be that they are float precision errors and are actually really close to 0?

**Comment by marcus1487** [comment]
It seems that the core issue here has been resolved. If further clarification concerning negative mod_probs values arise, please open a new issue.
